### PR TITLE
Ts2

### PR DESCRIPTION
**Pull request by sharksystems**

### CODE REVIEW

File: src/main/java/base/TestInit.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.testng.annotations.BeforeMethod;
+import com.codeborne.selenide.Configuration;
+import static com.codeborne.selenide.Selenide.*;
+
+public class TestInit {
+
+//    public WebDriver driver;
+    @BeforeMethod
+    public void setup (){
+//        WebDriverManager.chromedriver().setup();
```
Comment (samsometro) [inline src/main/java/base/TestInit.java]: Clean code - delete all comments

File: src/main/java/pages/LoginPage.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.SelenideElement;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import pages.PageBase;
+import static com.codeborne.selenide.Selenide.$;
+
+public class LoginPage extends PageBase {
+//    WebDriver driver;
```
Comment (samsometro) [inline src/main/java/pages/LoginPage.java]: Same

File: src/main/java/pages/PageBase.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import com.codeborne.selenide.SelenideElement;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import static com.codeborne.selenide.Selenide.$;
+
+public class PageBase {
+//    WebDriver driver;
+//    public PageBase (WebDriver driver) {this.driver = driver;}
+
+    public final By homeBtn = By.xpath("//button[@class='btn home']");
```
Comment (samsometro) [inline src/main/java/pages/PageBase.java]: Create a new page for this locators and methods. Landing or Main Page must be more appropriate

File: src/test/java/tests/Test1.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import org.testng.Assert;
+import pages.LoginPage;
+import org.testng.annotations.Test;
+import static com.codeborne.selenide.Selenide.*;
+
+
+public class Test1 extends base.TestInit {
+    @Test
+    public void checkLoginPage() {
+        LoginPage loginPage = new LoginPage();
```
Comment (samsometro) [inline src/test/java/tests/Test1.java]: no need to create a variable for class object

File: src/test/java/tests/Test1.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        loginPage.findManagerLoginBtn();
+
+        Assert.assertTrue(loginPage.findCustomerLoginBtn().isDisplayed());
+        Assert.assertTrue(loginPage.findCustomerLoginBtn().isEnabled());
+        Assert.assertTrue(loginPage.findManagerLoginBtn().isDisplayed());
+        Assert.assertTrue(loginPage.findManagerLoginBtn().isEnabled());
+    }
+    @Test
+    public void checkMainHeader (){
+        LoginPage loginPage = new LoginPage();
```
Comment (samsometro) [inline src/test/java/tests/Test1.java]:
```suggestion
        new LoginPage();
```

File: src/test/java/tests/Test1.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import org.testng.annotations.Test;
+import static com.codeborne.selenide.Selenide.*;
+
+
+public class Test1 extends base.TestInit {
+    @Test
+    public void checkLoginPage() {
+        LoginPage loginPage = new LoginPage();
+        goToSite();
+        loginPage.findCustomerLoginBtn();
```
Comment (samsometro) [inline src/test/java/tests/Test1.java]: just call methods without variables

File: src/test/java/tests/Test1.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        loginPage.findManagerLoginBtn();
+
+        Assert.assertTrue(loginPage.findCustomerLoginBtn().isDisplayed());
```
Comment (samsometro) [inline src/test/java/tests/Test1.java]: Try static import

**Comment by sharksystems** [comment]
done

File: src/test/java/tests/Test1.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+import static pages.MainHeadPage.*;
+import org.testng.annotations.Test;
+import pages.MainHeadPage;
+import static com.codeborne.selenide.Selenide.*;
+
+
+public class Test1 extends base.TestInit {
+    @Test
+    public void checkLoginPage() {
+        new LoginPage();
```
Comment (samsometro) [inline src/test/java/tests/Test1.java]:
```suggestion
        new LoginPage()
      .goToSite()
      .findCustomerLoginBtn()
      .findManagerLoginBtn();
```

File: src/test/java/tests/Test1.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        goToSite();
+        findCustomerLoginBtn();
```
Comment (samsometro) [inline src/test/java/tests/Test1.java]:
```suggestion

```

File: src/test/java/tests/Test1.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+        findManagerLoginBtn();
```
Comment (samsometro) [inline src/test/java/tests/Test1.java]:
```suggestion
 
```

File: src/test/java/tests/Test1.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+        Assert.assertTrue(findCustomerLoginBtn().isDisplayed());
+        Assert.assertTrue(findCustomerLoginBtn().isEnabled());
+        Assert.assertTrue(findManagerLoginBtn().isDisplayed());
+        Assert.assertTrue(findManagerLoginBtn().isEnabled());
+    }
+    @Test
+    public void checkMainHeader (){
+        new MainHeadPage();
+        goToSite();
+        findMainHeadingTxt();
```
Comment (samsometro) [inline src/test/java/tests/Test1.java]: You already have assertions. Why do you call this methods in the test than?

File: src/main/java/pages/MainHeadPage.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+import org.openqa.selenium.By;
+
+import static com.codeborne.selenide.Selenide.$;
+
+public class MainHeadPage extends PageBase {
+    public static final By homeBtn = By.xpath("//button[@class='btn home']");
+    public static final By logoutBtn = By.xpath("//button[@class='btn logout']");
+    public static final By mainHeadingTxt = By.xpath("//strong[@class='mainHeading']");
+
+
```
Comment (samsometro) [inline src/main/java/pages/MainHeadPage.java:16]:
```suggestion
```

File: src/main/java/pages/MainHeadPage.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    public static void clickHomeBtn () {$(homeBtn).shouldBe(Condition.visible).click();}
+    public static void clickLogoutBtn () {$(logoutBtn).shouldBe(Condition.visible).click();}
+    public static SelenideElement findMainHeadingTxt () { return $(mainHeadingTxt).shouldBe(Condition.visible);}
```
Comment (samsometro) [inline src/main/java/pages/MainHeadPage.java]:
```suggestion
    public SelenideElement findMainHeadingTxt () { return $(mainHeadingTxt).shouldBe(Condition.visible);}
```

File: src/main/java/pages/MainHeadPage.java [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+
+import static com.codeborne.selenide.Selenide.$;
+
+public class MainHeadPage extends PageBase {
+    public static final By homeBtn = By.xpath("//button[@class='btn home']");
+    public static final By logoutBtn = By.xpath("//button[@class='btn logout']");
+    public static final By mainHeadingTxt = By.xpath("//strong[@class='mainHeading']");
+
+
+    public static void clickHomeBtn () {$(homeBtn).shouldBe(Condition.visible).click();}
```
Comment (samsometro) [inline src/main/java/pages/MainHeadPage.java]: same

File: src/main/java/pages/MainHeadPage.java [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+    public static void clickLogoutBtn () {$(logoutBtn).shouldBe(Condition.visible).click();}
```
Comment (samsometro) [inline src/main/java/pages/MainHeadPage.java]: same

File: src/main/java/pages/MainHeadPage.java [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+    public static SelenideElement findMainHeadingTxt () { return $(mainHeadingTxt).shouldBe(Condition.visible);}
+    public static SelenideElement findHomeBtn () { return $(homeBtn).shouldBe(Condition.visible);}
```
Comment (samsometro) [inline src/main/java/pages/MainHeadPage.java]: I think U got me)

File: src/main/java/pages/PageBase.java [hunk 17]
```diff
@@ -0,0 +1,10 @@
+package pages;
+
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.SelenideElement;
+import org.openqa.selenium.By;
+import static com.codeborne.selenide.Selenide.$;
+
+public class PageBase {
+    //Nobody here but us mice...
```
Comment (samsometro) [inline src/main/java/pages/PageBase.java:6]: Clever)

File: src/main/java/pages/MainHeadPage.java [hunk 18]
```diff
@@ -0,0 +1,20 @@
+package pages;
+
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.SelenideElement;
+import org.openqa.selenium.By;
+
+import static com.codeborne.selenide.Selenide.$;
+
+public class MainHeadPage extends PageBase {
+    public static final By homeBtn = By.xpath("//button[@class='btn home']");
```
Comment (samsometro) [inline src/main/java/pages/MainHeadPage.java:11]: навіщо тут static?

File: src/test/java/tests/Test1.java [hunk 19]
```diff
@@ -0,0 +1,37 @@
...
+public class Test1 extends base.TestInit {
+    @Test
+    public void checkLoginPage() {
+        LoginPage loginPage = new LoginPage();
+        goToSite();
+        loginPage.findCustomerLoginBtn();
+        loginPage.findManagerLoginBtn();
+
+        Assert.assertTrue(loginPage.findCustomerLoginBtn().isDisplayed());
+        Assert.assertTrue(loginPage.findCustomerLoginBtn().isEnabled());
```
Comment (samsometro) [inline src/test/java/tests/Test1.java]:
```suggestion
        assertTrue(loginPage.findCustomerLoginBtn().isEnabled());
```

Comment (samsometro) [reply src/test/java/tests/Test1.java]: Я мав на увазі цей статік імпорт)

**Review by samsometro — APPROVED** [review]

**Pull request merged**